queue_processors/queue_processor/queue_processor.py

```diff
@@ -72,7 +72,10 @@ def on_message(self, headers, message):
         logger.info("Destination: %s Priority: %s", destination, self._priority)
         # Load the JSON message and header into dictionaries
         try:
-            self._data_dict = json.loads(message)
+            if isinstance(message, Message):
```

Comment:
This is fine to have, although I'm not sure we would ever hit this statement. Still might save us a run failure at some point